source/MaterialXGenShader/Nodes/ConvertNode.cpp

```diff
@@ -0,0 +1,135 @@
+#include <MaterialXGenShader/Nodes/ConvertNode.h>
+#include <MaterialXGenShader/ShaderGenerator.h>
+#include <MaterialXGenShader/TypeDesc.h>
+
+namespace MaterialX
+{
+
+namespace
+{
+    using ConvertTable = std::unordered_map<const TypeDesc*, std::unordered_map<const TypeDesc*, std::string> >;
+
+    static const ConvertTable CONVERT_TABLE({
```

Comment:
Table of swizzle patterns to use when converting from one type to another. Only the conversions supported according to spec is added.